recipe/meta.yaml

```diff
@@ -129,13 +130,25 @@ test:
     - tabulate
   imports:
     - torch
+  files:
+    - smoke_test.py
   source_files:
     - test/
   commands:
     # We seem to have individual platform-specific test failures or flaky
     # tests, but the majority of tests pass.
     - set CONTINUE_THROUGH_ERROR=1     # [win]
     - export CONTINUE_THROUGH_ERROR=1  # [not win]
+    # the smoke test script takes a bunch of env variables, defined below
+    - set MATRIX_GPU_ARCH_VERSION="{{ '.'.join(cudatoolkit.split('.')[:2]) }}"    # [(pytorch_variant == "gpu") and (win)]
+    - set MATRIX_GPU_ARCH_TYPE="cuda"                                             # [(pytorch_variant == "gpu") and (win)]
+    - set MATRIX_GPU_ARCH_VERSION="none"                                          # [(pytorch_variant == "cpu") and (win)]
+    - set MATRIX_GPU_ARCH_TYPE="none"                                             # [(pytorch_variant == "cpu") and (win)]
+    - set MATRIX_CHANNEL="defaults"                                               # [win]
+    - set MATRIX_STABLE_VERSION={{ version }}                                     # [win]
+    - set MATRIX_PACKAGE_TYPE="conda"                                             # [win]
+    - set TARGET_OS="windows"                                                     # [win]
+    - python ./smoke_test.py --package torchonly
```

Comment:
all the env variables above are used by the smoke_test for win, so that smoke_test shouldn't be windows only, too ?

Reply:
The other environment variables were added by a subsequent PR for pytorch 2.1. My changes were made to make it easy for them to be integrated in Daniel's work without creating nasty merge conflicts.